### ISSUE TITLE
Kong 2.8.1 vs 3.2.2 : Performance Degrade in Kong 3.2.2 when compared to 2.8.1 

### ISSUE THREAD

**Issue by 1madhu9** [Kong/kong#10844]
We have executed a performance test against kong 3.2.2 and were observing increase in kong latency and also decrease in throughput when compared to kong 2.8.1 version 

We had executed a test with a proxy(service/route pair) running with request termination plugin enabled on the service returning a static plain text message and HTTP 200 on two different Kong versions (v 2.8.1 and 3.2.2) and we do see a significant difference in the throughput and proxy latency of these 2 versions.

<img width="919" alt="image" src="https://github.com/Kong/kong/assets/130395486/99a450da-a7ec-493d-bc01-3a9774ab0c2c">

### Expected Behaviour
Do not observe degrade in proxy latency when upgraded to a latest version

### Steps To Reproduce
Run load test as prescribed above on the given Kong versions.

### Environment details 
Deployed our own kong docker image on Alpine Base
Using postgres as backend DB

**Comment by bungle** [comment]
@1madhu9, can you help us on how we could reproduce this test. We multiple tests on our own, but we would like to know the test setup that you had.

**Comment by 1madhu9** [comment]
@bungle, We are building Kong image on Alpine base and add some of the custom plugins that are built internally.

Kong is deployed as an ECS service in AWS with the following configuration:
- 6 instances of type c5.2xLarge.
- 2 Tasks running in each instance, each containing the Kong proxy container.
- Load Balancing - AWS ELB to load balance traffic to kong instances 

For the test, we utilized the 'hey' load testing tool.
We set up 5 Load Generators (LGs) of type t3.small to generate a concurrent user load of 200 users (40 users on each LG).
We executed 'hey' as a Docker image with the following sample command:

docker run --net=host --rm mesosphere/hey -z 720s -c 40 -H "Authorization: Bearer ${iamToken}" https://${gateway_host}/${service_entity}/health

- The duration of the test was set to 720 seconds.
- Each LG simulated 40 concurrent users.

Note: The network parameter was not involved in the test as we enabled the request termination plugin on the service entity. The plugin returns a static output message.
